#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_non_suse.dpatch by Petter Reinholdtsen <pere@hungry.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Avoid crashing with null-pointer when compiling without SUSE flag.

@DPATCH@
--- insserv-1.04.0.orig/insserv.c
+++ insserv-1.04.0/insserv.c
@@ -1984,7 +1984,7 @@
 	     * default_stop arn't used in SuSE Linux.
 	     */
 	    if (script_inf.default_stop && script_inf.default_stop != empty) {
-		if (!service || !del)
+		if (service && !del)
 		    service->lvlk = str2lvl(script_inf.default_stop);
 	    }
 #endif
